File: CarHomeWidget/src/main/java/com/anod/car/home/prefs/model/AppSettings.java
package com.anod.car.home.prefs.model;

import android.content.ComponentName;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.anod.car.home.CarWidgetApplication;
import com.anod.car.home.R;

/**
 * @author algavris
 * @date 19/03/2016.
 */
public class AppSettings extends ChangeableSharedPreferences {
    public static final String MUSIC_APP = "music-app";
    private static final String APP_THEME = "app_theme";

    public static AppSettings create(Context context) {
        return new AppSettings(context);
    }

    public AppSettings(Context context) {
        super(getSharedPreferences(context));
    }

    static SharedPreferences getSharedPreferences(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    public void setMusicApp(ComponentName musicApp) {
        putChange(MUSIC_APP, musicApp);
    }

    public ComponentName getMusicApp() {
        String musicAppString = mPrefs.getString(MUSIC_APP, null);

        if (musicAppString != null) {
            return ComponentName.unflattenFromString(musicAppString);
        }
        return null;
    }

    public int getTheme() {
        return mPrefs.getInt(APP_THEME, AppTheme.THEME_GRAY);
    }

    public void setAppTheme(int theme) {
        putChange(APP_THEME, theme);
    }

}

File: app/src/main/java/com/anod/car/home/prefs/model/AppSettings.kt
package com.anod.car.home.prefs.model

import android.content.ComponentName
import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager
import info.anodsplace.carwidget.preferences.model.ChangeableSharedPreferences

/**
 * @author algavris
 * @date 19/03/2016.
 */
class AppSettings(context: Context) : ChangeableSharedPreferences(getSharedPreferences(context)) {

    var musicApp: ComponentName?
        get() {
            val musicAppString = prefs.getString(MUSIC_APP, null)

            return if (musicAppString != null) {
                ComponentName.unflattenFromString(musicAppString)
            } else null
        }
        set(musicApp) = putChange(MUSIC_APP, musicApp)

    var theme: Int
        get() = prefs.getInt(APP_THEME, AppTheme.gray)
        set(value) = putChange(APP_THEME, value)

    companion object {
        const val MUSIC_APP = "music-app"
        private const val APP_THEME = "app_theme"

        private fun getSharedPreferences(context: Context): SharedPreferences {
            return PreferenceManager.getDefaultSharedPreferences(context)
        }
    }

}
